op-devstack/sysgo/superroot.go

```diff
@@ -0,0 +1,445 @@
+	t devtest.CommonT,
+	transactChainID *big.Int,
+	privateKey *ecdsa.PrivateKey,
+	client *ethclient.Client,
+	delegateCallProxy common.Address,
+	proxyAdminOwned common.Address,
+	newOwner common.Address,
+) {
+	transactOpts, err := bind.NewKeyedTransactorWithChainID(privateKey, transactChainID)
+	t.Require().NoError(err, "must have transact opts")
+	transactOpts.Context = t.Ctx()
+
+	abi, err := delegatecallproxy.DelegatecallproxyMetaData.GetAbi()
+	t.Require().NoError(err, "failed to get abi")
+	contract := batching.NewBoundContract(abi, delegateCallProxy)
+	call := contract.Call("transferOwnership", proxyAdminOwned, newOwner)
+	data, err := call.Pack()
+	t.Require().NoError(err)
+
+	candidate := txmgr.TxCandidate{
+		To:       &delegateCallProxy,
+		TxData:   data,
+		GasLimit: 1_000_000,
+	}
+	_, receipt, err := transactions.SendTx(t.Ctx(), client, candidate, privateKey)
+	t.Require().NoErrorf(err, "transferOwnership failed: %v", errutil.TryAddRevertReason(err))
+	t.Require().Equal(receipt.Status, types.ReceiptStatusSuccessful, "transferOwnership failed")
+}
+
+func changeAdminForDelegateCallProxy(
+	t devtest.CommonT,
+	transactChainID *big.Int,
+	privateKey *ecdsa.PrivateKey,
+	client *ethclient.Client,
+	delegateCallProxy common.Address,
+	proxyAdminOwned common.Address,
+	newOwner common.Address,
+) {
+	transactOpts, err := bind.NewKeyedTransactorWithChainID(privateKey, transactChainID)
+	t.Require().NoError(err, "must have transact opts")
+	transactOpts.Context = t.Ctx()
+
+	abi, err := delegatecallproxy.DelegatecallproxyMetaData.GetAbi()
+	t.Require().NoError(err, "failed to get abi")
+	contract := batching.NewBoundContract(abi, delegateCallProxy)
+	call := contract.Call("changeAdmin", proxyAdminOwned, newOwner)
+	data, err := call.Pack()
+	t.Require().NoError(err)
+
+	candidate := txmgr.TxCandidate{
+		To:       &delegateCallProxy,
+		TxData:   data,
+		GasLimit: 1_000_000,
+	}
+	_, receipt, err := transactions.SendTx(t.Ctx(), client, candidate, privateKey)
+	t.Require().NoErrorf(err, "changeAdmin failed: %v", errutil.TryAddRevertReason(err))
+	t.Require().Equal(receipt.Status, types.ReceiptStatusSuccessful, "changeAdmin failed")
+}
+
+func resetOwnershipAfterMigration(
+	t devtest.CommonT,
+	o *Orchestrator,
+	l1ChainID *big.Int,
+	ownerPrivateKey *ecdsa.PrivateKey,
+	w3Client *w3.Client,
+	client *ethclient.Client,
+	delegateCallProxy common.Address,
+	opChainConfigs []bindings.OPContractsManagerOpChainConfig,
+) {
+	l1PAO, err := o.keys.Address(devkeys.ChainOperatorKeys(l1ChainID)(devkeys.L1ProxyAdminOwnerRole))
+	t.Require().NoError(err, "must have L1 proxy admin owner private key")
+
+	portal0 := getOptimismPortal(t, w3Client, opChainConfigs[0].SystemConfigProxy)
+	sharedDGF := getDisputeGameFactory(t, w3Client, portal0)
+	transferOwnershipForDelegateCallProxy(
+		t,
+		l1ChainID,
+		ownerPrivateKey,
+		client,
+		delegateCallProxy,
+		sharedDGF,
+		l1PAO,
+	)
+
+	// TODO(#16037): fix ownership reset of new proxies
```

Comment:
I wonder if we actually need to reset ownership or if we should just record the new owner.  If we need to make other calls we'd likely need to jump through the same delegate proxy hoops anyway.

Reply:
There aren't any tests today that care what the L1PAO. However, it's not obvious that this is happening and I could see a test added later (like a standard validator test) that won't be compatible with the new owner. So I think we should ensure that the owner is consistent, what ever that may be. It's easier to reset ownership to the original account, rather than have a state where some proxies are owned by the new L1PAO but not all.